pkg/collector/corechecks/ebpf/probe/oom_kill.go

```diff
@@ -67,13 +104,10 @@ func (k *OOMKillProbe) Get() []OOMKillStats {
 	}
 
 	var results []OOMKillStats
-
-	for it := k.oomMap.Iter(); it.Next(); {
-		var stat C.struct_oom_stats
-
-		data := it.Leaf()
-		C.memcpy(unsafe.Pointer(&stat), unsafe.Pointer(&data[0]), C.sizeof_struct_oom_stats)
-
+	var pid uint32
+	var stat C.struct_oom_stats
+	it := k.oomMap.Iterate()
+	for it.Next(&pid, &stat) {
```

Comment:
I think you also need to `it.Err()` after the loop.

Reply:
I believe this should be `unsafe.Pointer(&pid)`.

Reply:
uint32 is a known type to the library. Used unsafe.Pointer for the second one